pkg/kubectl/cmd/apply.go

```diff
@@ -61,6 +61,8 @@ const (
 	backOffPeriod = 1 * time.Second
 	// how many times we can retry before back off
 	triesBeforeBackOff = 1
+
+	warningNoLastAppliedConfigAnnotation = "Warning: kubectl apply should be used on resource created by eithor kubectl create --save-config or kubectl apply\n"
```

Comment:
Yes, it would be breaking. And if this were implemented as an error, there should be a dedicated `kubectl apply`-specific command-line flag which allows you to always continue anyways and just generate a new annotation in the event that there isn't one already present.